v2/apiserver/internal/core/events.go

```diff
@@ -812,6 +835,49 @@ func (e *eventsService) DeleteMany(
 	return result, nil
 }
 
+func (e *eventsService) Retry(
+	ctx context.Context,
+	id string,
+) (Event, error) {
+	// No authz call here as we'll defer to the checks in e.Create() invoked
+	// below
+
+	event, err := e.eventsStore.Get(ctx, id)
+	if err != nil {
+		return Event{}, errors.Wrapf(
+			err,
+			"error retrieving event %q from store",
+			id,
+		)
+	}
+
+	// Only allow retry if the event Worker has reached a terminal phase
+	if !event.Worker.Status.Phase.IsTerminal() {
```

Comment:
I think it would be appropriate to return a `*meta.ErrConflict` here. The REST machinery ultimately translates that to a 409.